src/node_crypto.cc

```diff
@@ -2563,6 +2564,18 @@ static bool IsSupportedAuthenticatedMode(int mode) {
          IS_OCB_MODE(mode);
 }
 
+static bool IsAEADChacha20Poly1305(const EVP_CIPHER* cipher) {
+  // see openssl/crypto/evp/e_chacha20_poly1305.c#L612
+  const int CHACHA20_POLY1305_FLAG = EVP_CIPH_FLAG_AEAD_CIPHER
+                                   | EVP_CIPH_CUSTOM_IV
+                                   | EVP_CIPH_ALWAYS_CALL_INIT
+                                   | EVP_CIPH_CTRL_INIT
+                                   | EVP_CIPH_CUSTOM_COPY
+                                   | EVP_CIPH_FLAG_CUSTOM_CIPHER;
+  return EVP_CIPHER_flags(cipher) == CHACHA20_POLY1305_FLAG
+      && EVP_CIPHER_nid(cipher) == NID_chacha20_poly1305;
```

Comment:
Why isn't the second check enough? Why do we need to check that the flags are an exact match?

Reply:
I think nid is strong enough, will fix it